services/src/main/java/org/jboss/windup/web/services/rest/UnhandledExceptionHandler.java

```diff
@@ -62,15 +62,15 @@ else if (null != findFirstOccurenceInExceptionChain("org.h2.jdbc.JdbcSQLExceptio
 
     static final Throwable findFirstOccurenceInExceptionChain(Class<? extends Throwable> type, Throwable exception)
     {
-        Throwable cause = exception;
+        Throwable cause = exception.getCause();
```

Comment:
You are right guys, it didn't really solve those issues. I will write some tests and fix it.